### PR TITLE
AME Deconstruction Changes

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

deconstructing damaged AME shielding now yields 5 steel instead of a full AME shield flatpack.
if you run a normal ass AME with no overclocking you'll notice no change.
If you overclock an AME, then the cores will yield steel once the controller switches to the warning state.

## Why / Balance

being able to reconstruct the AME shielding to reset the integrity is extremely wack. it nullifies the whole point of the damage aspect when you can just reset it in 3 seconds. At the same time, the deconstruction is important because lord knows many an engineer has sealed themselves inside the AME tomb.

this basically just lets normal AME activities continue as normal while limiting the extent to which someone can overclock an AME to extend power infinitely.

## Technical details

adds a nice lil construction condition that nothing else will ever use.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Deconstructing a damaged AME core now yields steel instead of a full AME flatpack.

### CODE REVIEW

File: Content.Server/Construction/Conditions/AmeShieldIntegrity.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using Content.Shared.Construction;
+using JetBrains.Annotations;
+using Content.Shared.Examine;
+
+namespace Content.Server.Construction.Conditions;
+
+    [UsedImplicitly]
+    [DataDefinition]
+    public sealed partial class AmeShieldIntegrity : IGraphCondition
+    {
```
Comment (osjarw) [inline Content.Server/Construction/Conditions/AmeShieldIntegrity.cs:11]: Are we supposed to indent classes if we are using file-scoped namespaces?

**Comment by Cojoke-dot** [comment]
Thiefs have the objective of stealing shielding, how do they complete it if Engi sets up the AME roundstart?

**Comment by thebadman4662** [comment]
Is even a single point of damage enough to turn it into steel? That would mean you can destroy AME silently and permanently in 16-6 seconds.

File: Content.Server/Construction/Conditions/AmeShieldIntegrity.cs (hunk 1 above)
Comment (Brandon-Huu) [reply Content.Server/Construction/Conditions/AmeShieldIntegrity.cs:11]: Hell naw

**Comment by Brandon-Huu** [comment]
> Is even a single point of damage enough to turn it into steel? That would mean you can destroy AME silently and permanently in 16-6 seconds.

Could also just deconstruct and steal it

**Comment by thebadman4662** [comment]
If you just steal it you have to carry it and risk it getting found, if you wreck core into steel you do not need to worry about getting caught and it getting back up.

**Comment by osjarw** [comment]
> If you just steal it you have to carry it and risk it getting found, if you wreck core into steel you do not need to worry about getting caught and it getting back up.

stealing it should take like 4 seconds for a single core AME, and I doubt that anyone stealing it would do so without further plans for disposing of the shielding

> Is even a single point of damage enough to turn it into steel? That would mean you can destroy AME silently and permanently in 16-6 seconds.

The shielding integrity needs to go below 80 before it turns into steel. That means that the AME needs to be OC'd for at least 50 seconds before it can be scrapped.

**Comment by XxSWAG-MASTER-OVERLORDxX** [comment]
Won’t this make getting the AME part for thief’s much harder/ impossible?

**Comment by thebadman4662** [comment]
> Won’t this make getting the AME part for thief’s much harder/ impossible?

Only inner AME takes damage, other 8 shielding parts are 100/100.

If map comes with more than 9AME parts yous till can OC ame for long ass time.

**Pull request merged**